localstack/services/dynamodb/provider.py

```diff
@@ -294,6 +294,30 @@ def get_store(context: RequestContext | None = None) -> DynamoDBStore:
     return dynamodb_stores[_account_id][_region]
 
 
+def get_global_table_region(table_name: str, target_region: str | None = None) -> str | None:
+    """
+    Check if the table is a Version 2019.11.21 table.
+    :param target_region: the region we are looking for
+    :param table_name: the name of the table
+    :return: the original region; `None` if this is not a global table
+    """
+    replicas = get_store().REPLICA_UPDATES.get(table_name)
+    if not replicas:
+        return None
+    # todo: consider a different data structure to keep the list of replicas
+    original_region = list(replicas.keys())[0]
+    return original_region if target_region in replicas[original_region] else None
+
+
+def look_for_global_table(
+    context: RequestContext, table_name: str, region: str | None = None
+) -> None:
+    region: str | None = get_global_table_region(table_name=table_name, target_region=region)
+    if region:
+        # modify the context to query the original region where the table has been created
+        context.region = region
```

Comment:
modifying the request context this way can lead to unexpected behavior downstream in the handler chain. i'm not sure it will _really_ be a problem. maybe it would be better to create a fresh request context and make a completely new request transparently to the client. but then the response would contain a region that's different to the original request, which may be equally problematic :shrug:

i guess we can live with it for now. any maybe @alexrashed has some thoughts about it.

Reply:
I don't think it causes problems directly, but I do agree that these properties of the context should be somewhat immutable.
Imho, these modifications should be handled by creating a new context which is only created for the forwarding.
In fact, the `ForwardingFallbackDispatcher` already creates a separate context for the request forwarding (in most cases):
https://github.com/localstack/localstack/blob/4a58a975d8549a7f4c83899406258a911ddc5114/localstack/aws/forwarder.py#L70-L79

So the cleanest way would be to refactor the forwarding such that it becomes a bit more flexible, but _currently_ the context modification should not cause any issues.